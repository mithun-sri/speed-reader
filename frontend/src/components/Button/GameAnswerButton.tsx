import { Box, Icon, IconButton } from "@mui/material";
import { useEffect, useState } from "react";


const GameAnswerButton: React.FC<{
    text: string;
}> = ({ text }) => {
   const [fontSize, setFontSize] = useState(calculateFontSize());
   const [innerWidth, setInnerWidth] = useState(calculateWidthSize());

   useEffect(() => {
     function handleResize() {
        setInnerWidth(calculateWidthSize());
       setFontSize(calculateFontSize());
     }

      window.addEventListener("resize", handleResize);
      return () => {
       window.removeEventListener("resize", handleResize);
     };
   }, []);
  
   function calculateFontSize() {
     const windowWidth = window.innerWidth;
     const minSize = 12;
     const maxSize = 36;
      const calculatedSize = Math.min(
       maxSize,
       Math.max(minSize, windowWidth / 15)
     );
      return calculatedSize;
   }

   function calculateWidthSize() {
    const windowWidth = window.innerWidth;
    const minWidthPercent = 40; // Set a minimum width as a percentage of the window width
    const maxWidthPercent = 80; // Set a maximum width as a percentage of the window width
    const borderWidth = 20; 
    const paddingWidth = 40;
  
    // Calculate the maximum available width for the box content as a percentage of the window width
    const maxContentWidth = (maxWidthPercent / 100) * windowWidth - borderWidth - paddingWidth;
  
    // Ensure the box content width is at least the minimum
    const calculatedSize = Math.max((minWidthPercent / 100) * windowWidth, Math.min(maxContentWidth, windowWidth / 20));
  
    // Ensure the entire box (including border and padding) fits within the window width
    return Math.min(calculatedSize, windowWidth - borderWidth);
  }

   return (
       <IconButton sx={{
           fontFamily: "JetBrains Mono, monospace",
           color: "#FFFFFF"
       }}>
           <Box sx={{
               width: innerWidth,
               border: '10px solid #646669',
               borderRadius: "30px",
               background: "#E2B714",
               padding: "10px 20px 10px 20px",
               fontWeight: "bolder",
               fontSize: fontSize / 1.15,
               wordWrap: "break-word",
               textAlign: "center"
           }}>
               <Box>{text}.</Box>
           </Box>
       </IconButton>
   );
}


export default GameAnswerButton;
