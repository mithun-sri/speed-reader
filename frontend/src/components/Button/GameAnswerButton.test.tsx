import { faker } from "@faker-js/faker";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import GameAnswerButton from "./GameAnswerButton";

it("Matches snapshot", () => {
  const text = faker.lorem.word(2);
  const component = renderer.create(<GameAnswerButton text={text} />);
  const tree = component.toJSON() as ReactTestRendererJSON;
  expect(tree).toMatchSnapshot();
});
