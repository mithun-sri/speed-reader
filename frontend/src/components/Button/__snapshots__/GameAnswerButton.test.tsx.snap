// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-11jexo3-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiBox-root css-1dkfg4q"
  >
    <div
      className="MuiBox-root css-0"
    >
      at
      .
    </div>
  </div>
</button>
`;
